### PR TITLE
Bug #15216: Fix a link for open files in IDE

### PR DESCRIPTION
**Pull request by asamats**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #15216

### CODE REVIEW

**Comment by samdark** [comment]
@asamats would you please add a line for changelog?

**Comment by asamats** [comment]
@samdark Yes, I can and must. Thanks.

File: framework/views/errorHandler/callStackItem.php [hunk 1]
```diff
@@ -9,13 +9,28 @@
...
 /* @var $args array */
 /* @var $handler \yii\web\ErrorHandler */
+$html = <<<HTML
+IDE
+<svg class="icon icon--new-window" focusable="false" aria-hidden="true" width="16" height="16">
+    <use href="#new-window"></use>
+</svg>
+HTML;
 ?>
+<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" hidden>
```
Comment (samdark) [inline framework/views/errorHandler/callStackItem.php]: What is this SVG for? I don't quite understand how does that work...

**Review by samdark — CHANGES_REQUESTED** [review]

File: framework/views/errorHandler/callStackItem.php (hunk 1 above)
Comment (asamats) [reply framework/views/errorHandler/callStackItem.php]: I added an icon for an external link. It's just UI/UX thing.

It's "link" by id #new-window. I did this way to save a little bit of data/DOM size.
Info: https://www.w3.org/TR/SVG/linking.html

Excuse me, if I've not understood your question.

**Comment by samdark** [comment]
Doesn't work well to me. The icon is there but there's unexpected spacing introduced in Chrome-based browsers:

c1

It's more visible if scrolling down a bit:

c2

File: framework/views/errorHandler/callStackItem.php [hunk 2]
```diff
@@ -9,13 +9,28 @@
...
+        </g>
+    </symbol>
+</svg>
 <li class="<?= ($index === 1 || !$handler->isCoreFile($file)) ? 'application' : '' ?> call-stack-item"
     data-line="<?= (int) ($line - $begin) ?>">
     <div class="element-wrap">
         <div class="element">
             <span class="item-number"><?= (int) $index ?>.</span>
             <span class="text"><?= $file !== null ? 'in ' . $handler->htmlEncode($file) : '' ?></span>
+            <span> - </span>
```
Comment (alexantr) [inline framework/views/errorHandler/callStackItem.php]: Need `&ndash;` like in `<span class="call">` below

Comment (asamats) [reply framework/views/errorHandler/callStackItem.php]: That's not the problem. The problem is svg element. It has height, that is why we have this problem.
Problem solution:
1) svg must be invisible
2) svg must location higher

Comment (alexantr) [reply framework/views/errorHandler/callStackItem.php]: Remove svg at all or use base64 png like links in the header

Comment (samdark) [reply framework/views/errorHandler/callStackItem.php]: SVG is fine if it behaves well.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**